### PR TITLE
Multiple adaptions for quick fixes

### PR DESCRIPTION
**Pull request by dennismeister93**
Add delay for waiting of git api. Add already change for future event loop of template

Signed-off-by: Dennis Meister <dennis.meister@bosch.com>

### CODE REVIEW

File: src/project-generator.ts [hunk 1]
```diff
@@ -67,6 +71,9 @@ export class ProjectGenerator {
     public async run(codeSnippet: string, appName: string): Promise<number> {
         try {
             await this.generateRepo();
+            // Delay is introduced to make sure that the git API creates
+            // everything we need before doing other API requests
+            await this.delay(5000);
```
Comment (doosuu) [inline src/project-generator.ts]: Introduce timeout constant and use the same constant everywhere or is there a real reason for 5000ms vs 4000ms?

File: src/project-generator.ts [hunk 2]
```diff
@@ -167,6 +174,7 @@ export class ProjectGenerator {
             const newTreeSha = await this.createNewTreeSha(sha1, sha2, baseTreeSha);
             const mainBranchSha = await this.getMainBranchSha();
             const newCommitSha = await this.createCommitSha(mainBranchSha, newTreeSha);
+            await this.delay(4000);
```
Comment (doosuu) [inline src/project-generator.ts]: Move method to helpers and make it a function

**Review by doosuu — CHANGES_REQUESTED** [review]

File: src/project-generator.ts (hunk 1 above)
Comment (dennismeister93) [reply src/project-generator.ts]: Used same constant now and even reduced it a bit.

**Review by doosuu — APPROVED** [review]
LGTM

**Pull request merged**